.github/workflows/nightly-build.yml

```diff
@@ -0,0 +1,132 @@
+#===============================================================================
+# Copyright contributors to the oneDAL project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+name: Nightly-build
+
+on:
+  schedule:
+    - cron: '0 21 * * *'
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  build_lnx:
+    name: build Linux
+    if: github.repository == 'oneapi-src/oneDAL'
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+
+    steps:
+      - name: Checkout oneDAL
+        uses: actions/checkout@v4
+        with:
+          repository: oneapi-src/oneDAL
+      - name: Install DPC++
+        run: .ci/env/apt.sh dpcpp
+      - name: System Info
+        run: |
+          source /opt/intel/oneapi/compiler/latest/env/vars.sh
+          .ci/scripts/describe_system.sh
+      - name: Make daal
+        run: |
+          source /opt/intel/oneapi/compiler/latest/env/vars.sh
+          .ci/scripts/build.sh --compiler icx  --optimizations avx2 --target daal
+      - name: Make onedal
+        run: |
+          source /opt/intel/oneapi/compiler/latest/env/vars.sh
+          .ci/scripts/build.sh --compiler icx  --optimizations avx2 --target onedal
+      - name: Archive build
+        uses: actions/upload-artifact@v4
+        with:
+          name: __release_lnx
+          path: ./__release_lnx_icx
+
+  build_win:
+    name: build Windows
+    if: github.repository == 'oneapi-src/oneDAL'
+    runs-on: windows-2022
+    timeout-minutes: 120
+
+    steps:
+      - name: Checkout oneDAL
+        uses: actions/checkout@v4
+        with:
+          repository: oneapi-src/oneDAL
+      - name: Install DPC++
```

Comment:
can we unify things with bat scripts used in azure? https://github.com/oneapi-src/oneDAL/blob/main/.ci/pipeline/ci.yml#L695

so we will be using same process regardless of pipeline?

Reply:
@napetrov thank you for the suggestion. i have converted aspects into powershell files and made adaptations to minimize hard drive usage. Unfortunately it looks like even then the DevOps runner memory is insufficient. A second worry is that it pushes VC build times close to 3 hours. I could solve this by doing a similar DPCPP compiler preparation in a special DevOps nightly runner generating a compiler artifact. That would remove the 45 min compiler install and may reduce the memory footprint, but i would need to learn a bit more about DevOps and would need to test. even then it may not work (not sure how much memory would be saved, whether the compiler would work on a separate DevOps runner). let me know your thoughts. I would lean towards a follow up task for this.